src/header.rs

```diff
@@ -632,7 +633,11 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
       }
 
       self.write_bit(fi.is_filter_switchable)?;
-      self.write(2, fi.default_filter as u8)?;
+      if fi.is_filter_switchable {
+        self.write(2, FilterMode::SWITCHABLE as u8)?;
```

Comment:
This differs from AV1 spec section [5.9.10 Interpolation filter syntax](https://aomediacodec.github.io/av1-spec/#interpolation-filter-syntax) where `interpolation_filter` is implied rather than signaled when `is_filter_switchable == 1`.

Reply:
Thanks for catching this. Yes. I did wrong. Don't need "self.write(2, FilterMode::SWITCHABLE as u8)?;", once "self.write_bit(fi.is_filter_switchable)?;" tells that the filter is switchable. Instead, each inter block encode which filter to use. The " interp_filter[ dir ]' in https://aomediacodec.github.io/av1-spec/#inter-block-mode-info-syntax

In fact, this commit is WIP and yet no chance to test itself, part of the lager work that actually is able to use best subpel interpolation filter based on RDO.